annotation-processor/src/main/java/org/hibernate/validator/ap/ConstraintAnnotationVisitor.java

```diff
@@ -6,9 +6,11 @@
  */
 package org.hibernate.validator.ap;
 
+import java.util.Collections;
 import java.util.List;
+import java.util.Map;
 import java.util.Set;
-
+import java.util.stream.Collectors;
```

Comment:
With @gsmet we've been contemplating to get out all your nice improvements around the annotation processor as part of a 5.4 release (the reason being that 6 is the reference implementation for BV 2.0, so it will take quite a while until it will be delivered in a Final version). The usage of Java 8 prohibits that though. So if you like the idea of releasing these changes in 5.4, could you change that to use traditional loops etc.?